src/Cache/Signature.php

```diff
@@ -73,4 +73,19 @@ public function equals(SignatureInterface $signature)
 
         return true;
     }
+
+    private static function utf8Encode(array $data)
+    {
+        if (!function_exists('utf8_encode') || !function_exists('mb_detect_encoding')) {
```

Comment:
True, but basically nobody runs into this particular issue, but then we make everyone load a dependency they don't need either because this isn't an issue for them, or because they don't need the polyfill.